zqd/handlers_test.go

```diff
@@ -398,32 +399,46 @@ func TestPostNDJSONLogs(t *testing.T) {
 			ndjsonio.Rule{"_path", "http", "http_log"},
 		},
 	}
-	_, client, done := newCore(t)
-	defer done()
-	const spaceName = "test"
 
-	_, err := client.SpacePost(context.Background(), api.SpacePostRequest{Name: spaceName})
-	require.NoError(t, err)
+	test := func(t *testing.T, input string) {
+		_, client, done := newCore(t)
+		defer done()
+		const spaceName = "test"
 
-	payloads := postSpaceLogs(t, client, spaceName, &tc, false, src1)
-	last := payloads[len(payloads)-1].(*api.TaskEnd)
-	assert.Equal(t, last.Type, "TaskEnd")
-	assert.Nil(t, last.Error)
+		_, err := client.SpacePost(context.Background(), api.SpacePostRequest{Name: spaceName})
+		require.NoError(t, err)
 
-	res := zngSearch(t, client, spaceName, "*")
-	const expected = "#0:record[_path:string,ts:time,uid:bstring]\n0:[http;2;CXY9a54W2dLZwzPXf1;]\n0:[http;1;CXY9a54W2dLZwzPXf1;]"
-	require.Equal(t, expected, strings.TrimSpace(res))
+		payloads := postSpaceLogs(t, client, spaceName, &tc, false, input)
+		last := payloads[len(payloads)-1].(*api.TaskEnd)
+		assert.Equal(t, last.Type, "TaskEnd")
+		assert.Nil(t, last.Error)
 
-	min, max := nano.Ts(1e9), nano.Ts(2e9)
-	info, err := client.SpaceInfo(context.Background(), spaceName)
-	require.NoError(t, err)
-	require.Equal(t, &api.SpaceInfo{
-		MinTime:       &min,
-		MaxTime:       &max,
-		Name:          spaceName,
-		Size:          80,
-		PacketSupport: false,
-	}, info)
+		res := zngSearch(t, client, spaceName, "*")
+		const expected = "#0:record[_path:string,ts:time,uid:bstring]\n0:[http;2;CXY9a54W2dLZwzPXf1;]\n0:[http;1;CXY9a54W2dLZwzPXf1;]"
```

Comment:
nit: maybe hoist this to the top next to `src` so its easy for people reading the test to compare the two?

Reply:
👍